### PR TITLE
Added new google_firebase_apple_app_config datasource

### PR DESCRIPTION
**Pull request by AlfatahB**
Added `google_firebase_apple_app_config` datasource
fixes https://github.com/hashicorp/terraform-provider-google/issues/6520

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
`google_firebase_apple_app_config` (beta)
```

### CODE REVIEW

**Comment by mitj04** [comment]
LGTM

**Comment by c2thorn** [comment]
Looks like the timeout is being exceeded for the apple app resource in the test. Not sure how it is working with other similar tests. Can we try lengthening[ the create timeout](https://github.com/GoogleCloudPlatform/magic-modules/blob/0ea3137ff2ca621d51cf60d8bf8247b76ffa2afe/mmv1/products/dialogflowcx/terraform.yaml#L18) of that resource?

**Comment by AlfatahB** [comment]
> Looks like the timeout is being exceeded for the apple app resource in the test. Not sure how it is working with other similar tests. Can we try lengthening[ the create timeout](https://github.com/GoogleCloudPlatform/magic-modules/blob/0ea3137ff2ca621d51cf60d8bf8247b76ffa2afe/mmv1/products/dialogflowcx/terraform.yaml#L18) of that resource?

I've tried to increase the create timeout but the test is still failing.

File: mmv1/products/firebase/terraform.yaml [hunk 1]
```diff
@@ -89,6 +89,8 @@ overrides: !ruby/object:Overrides::ResourceOverrides
       custom_import: templates/terraform/custom_import/self_link_as_name.erb
       custom_delete: templates/terraform/custom_delete/firebase_app_deletion_policy.erb
   AppleApp: !ruby/object:Overrides::Terraform::ResourceOverride
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 60
```
Comment (c2thorn) [inline mmv1/products/firebase/terraform.yaml:93]:
```suggestion
```

**Comment by c2thorn** [comment]
I believe I found the root cause - the debug log revealed that there was an upstream error: `Too many Apps on project`
I've removed leftover apps on the project and the timeout change. Tests should rerun

**Review by c2thorn — APPROVED** [review]

**Comment by c2thorn** [comment]
LGTM, is there anything else pending due to the Draft status?

**Comment by AlfatahB** [comment]
> LGTM, is there anything else pending due to the Draft status?

I kept this PR as a draft only because the test was failing. As this test is passing now, marking this PR as ready for review.

**Pull request merged**